### DISCUSSION TITLE
Is there any option to pre-process login string?

### DISCUSSION THREAD

**Discussion by agaletski** [ltb-project/self-service-password#796] [General]
Thanks for great work to all who supports this project.

For my case users used to use login word in email format but the system uses short login (without @my.domain.in part)
Is there any option to preprocess login word like next:

if login = username@my.domain.in - truncate it to -> username
if login = username -leave it unchanged - > username

In my understanding, this can be realised by applying some regex to received $login variable before passing it further.

Thanks for any help or idea.

**Comment by coudot** [comment]
Maybe you could just configure the LDAP filter to match both attributes in your directory.

If using AD, this could be:
```php
$ldap_filter = "(&(objectClass=user)(|(sAMAccountName={login})(userPrinicipalName={login})))";
